#!/bin/sh -e
## xpp_support.dpatch by Henrique M. Holschuh <hmh@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: support xpp as a default print option

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch}"

case "$1" in
       -patch) patch $patch_opts -p1 < $0;;
       -unpatch) patch $patch_opts -p1 -R < $0;;
        *)
                echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
                exit 1;;
esac

exit 0

@DPATCH@
diff -urNad hplip/base/utils.py /tmp/dpep.Ghr1HQ/hplip/base/utils.py
--- hplip/base/utils.py	2005-01-21 11:21:52.000000000 -0200
+++ hplip/base/utils.py	2005-01-21 11:21:52.000000000 -0200
@@ -680,6 +680,7 @@
     
 def deviceDefaultFunctions():
     # Print
+    cmd_print = ''
     path = which( 'kprinter' )
     if len(path) > 0:
         cmd_print = 'kprinter -P%PRINTER% --system cups'
@@ -689,8 +690,11 @@
         if len(path) > 0:
             cmd_print = 'gtklp -P%PRINTER%'
         else:
-            cmd_print = ''
-        
+            path = which( 'xpp' )
+
+            if len(path) > 0:
+                cmd_print = 'xpp -d %PRINTER%'
+
     # Scan
     path = which( 'xsane' )
     if len(path) > 0:
